gui/wxpython/gui_core/menu.py

```diff
@@ -426,3 +426,32 @@ def RemoveNonExistentFiles(self):
 
         self._filehistory.Save(self._config)
         self._config.Flush()
+
+    @staticmethod
+    def PrintListOfRecentWorkspaceFiles(app_name="main"):
+        """Print a list of wxGUI recent workspace files
+
+        :param str app_name: the name of the application for which you
+                             want to print the recent files, default is
+                             "main" wxGUI
+        """
+        import configparser
+        import platform
+
+        from grass.lib.gis import CONFIG_DIR
+
+        config_dir_path = os.getenv("HOME")
+        if platform.system() == "Windows":
+            config_dir_path = os.getenv("APPDATA")
+        recent_files = os.path.join(
+            config_dir_path,
+            CONFIG_DIR,
+            RecentFilesMenu.recent_files,
+        )
+
+        if os.path.exists(recent_files):
+            config = configparser.ConfigParser()
+            config.read(recent_files)
+            for i in config[app_name].values():
+                if os.path.exists(i):
+                    print(i)
```

Comment:
I assume there is almost 100% duplication with some other code elsewhere. That's not good.

A better way to implement this is a Python API which returns a list. The CLI related code can use the API and do the printing.